design-documents/message-queue/PHPEnqueueLibrary.md

```diff
@@ -0,0 +1,54 @@
+
+[SimpleClient Implementation Code](https://github.com/php-enqueue/simple-client/blob/master/SimpleClient.php)
+
+Another option is [Queue Interoperablity Library](https://github.com/queue-interop/queue-interop) which also supports Enqueue and provide different level of controls like calling receive, reject & acknowledge functions, see [Interop\Queue\Consumer.php](https://github.com/queue-interop/queue-interop/blob/master/src/Consumer.php) (both options are discussed in the evaluation table above); but **Queue/Interop** maybe more plausible option with Magento core, since it gives similar controls which we need in QueueInterface implementation.
+
+## Magento to PHP Enqueue Queue Configuration
+
+| Config File Name (Magento) | Description                                                  | Config File Name (PHP Enqueue) | Description                                                  |
+| -------------------------- | ------------------------------------------------------------ | ------------------------------ | ------------------------------------------------------------ |
+| env.php                    | It defines queue connections parameters, separately for both AMQP and DB/MySql | config.xml                     | [Magento 2 Enqueue Config Example](https://github.com/php-enqueue/magento2-enqueue/blob/master/etc/config.xml) |
+| queue.xml                  | Brokers definitions that processes topics (only required for MySQL, not for RabbitMQ) | N/A                            | N/A                                                          |
+| communication.xml          | It defines the Topics (their request types, either response type for RPC style synchronous calls or schema, and multiple callback handlers (class type and method). | N/A                            | N/A                                                          |
+| queue_consumer.xml         | It defines the consumer processes and their attributes (like queue name, connection, maxMessages, consumerInstance type and callback handler etc.)<br />ConsumerInstance type defines what kind of Consumer you want to use in Magento as defined in [Different Types of Consumers](MagentoMessageQueueDesign.md#Different-Types-of-Consumers) | N/A                            | N/A                                                          |
+| queue_topology.xml         | Defines the exchange where Topic is been published (Topic to Queue mapping) | N/A                            | N/A                                                          |
+| queue_publisher.xml        | It defines the publishers and its attributes (connection name and exchange) | N/A                            | N/A                                                          |
+
+
+
+### Open Questions?
+
+1. There are two places where callback handlers are defined for a Topic/Queue. One in communications.xml; and other in queue_consumer.xml, which one has priority over other? 
```

Comment:
So about this one
Answer on this question depends from Consumer implementation
1) `Magento\Framework\MessageQueue\Consumer` and `Magento\Framework\MessageQueue\BatchConsumer` - this one using Handlers from "queue_consumer.xml" is set. If there are no one set, then from "communications.xml"
2) `Magento\AsynchronousOperations\Model\MassConsumer` - This one use both. First from queue_consumer and then result will be transferred to handler from communications.xml

Reply:
thanks for the feedback, this is helpful